Gems/Atom/Tools/AtomToolsFramework/Code/Source/Document/AtomToolsDocumentMainWindow.cpp

```diff
@@ -150,29 +125,39 @@ namespace AtomToolsFramework
         });
 
         m_actionSaveAll = CreateActionAtPosition(m_menuFile, insertPostion, "Save A&ll", [this]() {
-            bool result = false;
-            AtomToolsDocumentSystemRequestBus::EventResult(
-                result, m_toolId, &AtomToolsDocumentSystemRequestBus::Events::SaveAllDocuments);
-            if (!result)
+
+            for (int index = 0; index < m_tabWidget->count(); index++)
             {
-                SetStatusError(tr("Document save all failed"));
+                if (!SaveDocument(GetDocumentTabId(index)))
+                {
+                    // Stop if there is any save failed or cancel
+                    break;
```

Comment:
continue;

Reply:
If we continue here it would be a different behavior than what Chris mentioned in #11388 

> And I think the Cancel button should abort the entire Save All operation, not just the current file.

Reply:
The purpose of the tools framework is to keep things as consistent as possible, ideally requiring as little custom code as possible, between tools. That being said, using continue wouldn't do anything in this case. I must have been reading the wrong block. 

What you mentioned elsewhere makes more sense. Selecting no when asked about saving the file should continue the loop. Selecting cancel should break the loop. That should be consistent across all of the save all, close all, close all except actions. We will have to consolidate code and make sure things are behaving consistently between the document system class and the main window class.

Reply:
```
for (const auto& documentId : GetOpenDocumentIds())
{
    if (!SaveDocument(documentId))
    {
        break;
    }
}
```